### ISSUE TITLE
`Illegal HTTP message` start on Github API

### ISSUE THREAD

**Issue by MasseGuillaume** [akka/akka#20408]
Hi, I'm sending a request to github api via akka-http (2.4.4).

It fails with: `Illegal HTTP message start`

But it works fine with curl
```
curl \
  -H "Authorization: token xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" \
  https://api.github.com/repos/mrumkovskis/query/readme -vvv

*   Trying 192.30.252.124...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to api.github.com (192.30.252.124) port 443 (#0)
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS handshake, Server hello (2):
{ [98 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [2577 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [333 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [70 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* Server certificate:
*    subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*    start date: Apr  8 00:00:00 2014 GMT
*    expire date: Apr 12 12:00:00 2017 GMT
*    subjectAltName: api.github.com matched
*    issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*    SSL certificate verify ok.
> GET /repos/mrumkovskis/query/readme HTTP/1.1
> Host: api.github.com
> User-Agent: curl/7.47.1
> Accept: */*
> Authorization: token xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
> 
< HTTP/1.1 301 Moved Permanently
< Server: GitHub.com
< Date: Thu, 28 Apr 2016 04:41:35 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 174
< Status: 301 Moved Permanently
< X-RateLimit-Limit: 5000
< X-RateLimit-Remaining: 4992
< X-RateLimit-Reset: 1461818825
< X-OAuth-Scopes: 
< X-Accepted-OAuth-Scopes: admin:repo_hook, delete_repo, read:repo_hook, repo, repo:status, repo_deployment, write:repo_hook
< Location: https://api.github.com/repositories/1592551/readme
< X-GitHub-Media-Type: github.v3
< Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
< Access-Control-Allow-Origin: *
< Content-Security-Policy: default-src 'none'
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-XSS-Protection: 1; mode=block
< Vary: Accept-Encoding
< X-Served-By: a474937f3b2fa272558fa6dc951018ad
< X-GitHub-Request-Id: C654FD74:1559F:20634C08:5721947E
< 
{ [174 bytes data]

100   174  100   174    0     0    948      0 --:--:-- --:--:-- --:--:--   950
* Connection #0 to host api.github.com left intact
HTTP/1.1 301 Moved Permanently
Server: GitHub.com
Date: Thu, 28 Apr 2016 04:41:35 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 174
Status: 301 Moved Permanently
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4992
X-RateLimit-Reset: 1461818825
X-OAuth-Scopes: 
X-Accepted-OAuth-Scopes: admin:repo_hook, delete_repo, read:repo_hook, repo, repo:status, repo_deployment, write:repo_hook
Location: https://api.github.com/repositories/1592551/readme
X-GitHub-Media-Type: github.v3
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Access-Control-Allow-Origin: *
Content-Security-Policy: default-src 'none'
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
Vary: Accept-Encoding
X-Served-By: a474937f3b2fa272558fa6dc951018ad
X-GitHub-Request-Id: C654FD74:1559F:20634C08:5721947E

{
  "message": "Moved Permanently",
  "url": "https://api.github.com/repositories/1592551/readme",
  "documentation_url": "https://developer.github.com/v3/#http-redirects"
}
```

you can get your token here: https://github.com/settings/tokens/new

**Comment by ktoso** [comment]
Thanks for reporting!

**Comment by MasseGuillaume** [comment]
```
protected final def startNewMessage(input: ByteString, offset: Int): StateResult = {
  if (offset < input.length) setCompletionHandling(CompletionIsMessageStartError)
  try parseMessage(input, offset)
  catch { case NotEnoughDataException ⇒ continue(input, offset)(startNewMessage) }
}
```

`offset < input.length` that's not a off by one ?

**Comment by johanandren** [comment]
Calling that URL I get `Illegal response header: Illegal 'strict-transport-security' header: Invalid input ';', expected 'EOI'` so I think this is a duplicate of #20271

**Comment by johanandren** [comment]
Gist with (not-really-)repeater: https://gist.github.com/johanandren/c55ee7483bd18bb534ef037f97e367f3

**Comment by ktoso** [comment]
I'll take this over

**Comment by ktoso** [comment]
This indeed seems to be duplicate of #20271 and is solved since Akka 2.4.5
